helper/funcs.go

```diff
@@ -394,6 +404,24 @@ func CleanEnvVar(s string, r byte) string {
 	return string(b)
 }
 
+// CleanFilename replaces invalid characters in filename
+func CleanFilename(filename string, replace string) string {
```

Comment:
Probably because of the recent CVE patches I worked on, I just thought of a terrifying scenario where a malicious job name can be used to attack an operator with a management token via overwriting files in the operator's laptop via directory traversal.

Let's constrain these filenames to the current directory by checking [`PathEscapesSandbox`](https://github.com/hashicorp/nomad/pull/9346/files#diff-03576b8a5031987035a75fd5d5bf450b1b9c18993d89daa1ad62b202af1bb9c0R548), using the current working directory as the sandbox.

Reply:
I see the fix added and that LGTM

Reply:
I added a PathEscapesSandbox check in writeBytes() so that all files will be verified.  It's probably not necessary, but I added the check to mkdir() as well.